### PR TITLE
Fix broken embedded refs

### PR DESCRIPTION
**Pull request by samskiter**
I've changed the way schema refs are resolved through schemastore to fix this.

### CODE REVIEW

**Comment by samskiter** [comment]
There's a lot of string garbling to get this right...

File: jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/SchemaStore.java [hunk 1]
```diff
@@ -76,21 +77,42 @@ protected URI removeFragment(URI id) {
...
+        String stringId = id.toString();
+        if (stringId.endsWith("#")) {
+            try {
+                id = new URI(stripEnd(stringId, "#"));
+            } catch (URISyntaxException e) {
+                throw new IllegalArgumentException("Bad path: " + stringId);
+            }
+        }
+            
+        if (selfReferenceWithoutParentFile(parent, path) || substringBefore(stringId, "#").isEmpty()) {
```
Comment (samskiter) [inline jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/SchemaStore.java]: prevents us trying to use the contentresolver on a non-absolute incomplete URI

File: jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/SchemaStoreTest.java [hunk 2]
```diff
@@ -76,6 +76,20 @@ public void createWithSelfRef() throws URISyntaxException {
...
+    public void createWithEmbeddedSelfRef() throws URISyntaxException {
+
+        URI schemaUri = getClass().getResource("/schema/embeddedRef.json").toURI();
+
+        SchemaStore schemaStore = new SchemaStore();
+        Schema topSchema = schemaStore.create(schemaUri);
+        Schema embeddedSchema = schemaStore.create(topSchema, "#/definitions/embedded");
+        Schema selfRefSchema = schemaStore.create(embeddedSchema, "#");
+
+        assertThat(topSchema, is(sameInstance(selfRefSchema)));
```
Comment (samskiter) [inline jsonschema2pojo-core/src/test/java/org/jsonschema2pojo/SchemaStoreTest.java]: @joelittlejohn This is the core thing that I think I have fixed here. Do you agree with this test?

**Comment by joelittlejohn** [comment]
This looks good to me. Does this change fix #542?

**Comment by samskiter** [comment]
Hmmm, looks like it would. I wouldn't expect it to have even worked 2 levels deep in that example

**Pull request merged**